app/models/competition.rb

```diff
@@ -2525,12 +2525,23 @@ def payments_enabled?
     competition_payment_integrations.exists?
   end
 
+  def connected_payment_integration_types
+    raw_types = self.competition_payment_integrations.pluck(:connected_account_type)
+    raw_types.map { |type| CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS.invert[type] }
+  end
+
+  def payment_integration_connected?(integration_name)
+    CompetitionPaymentIntegration.validate_integration_name!(integration_name)
+
+    self.competition_payment_integrations.send(integration_name.to_sym).exists?
```

Comment:
If I'm understanding correctly, the `send` here is calling a scope on `competition_payment_integrations` (eg, competition_payment_integrations.paypal)? And then we do `exists?` to check if there's anything returned by that scope

Reply:
> If I'm understanding correctly, the `send` here is calling a scope on `competition_payment_integrations` (eg, competition_payment_integrations.paypal)?

This part is correct

> And then we do `exists?` to check if there's anything returned by that scope

This is only half-correct. There will always be something returned by the scope (as long as the method name referencing the scope exists in the first place, which we make sure through `validate_integration_name!`). Then calling the `exists?` method checks whether a *non-zero amount of entries* exists. In other words, whether there is an integration of the said scope type in the database.